### PR TITLE
RFP Proposal: Reputation System by BUIDL Labs

### PR DESCRIPTION
**Pull request by sahilnanda1995**

### CODE REVIEW

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+1. Scope of the reputation system service: 
+    * Collaborate with FileCoin DevGrants team to: 
+        * Gain feedback on the metrics we have thought of. How relevant would they be for users now?  
+        * Gain awareness on the kinds of objective feedback users will give that can be verifiable from the service.
+    * Iron out reputation score logic that will be the single metric that all miners will be ranked upon. 
+    * Finalize the reputation system service requirements spec sheet.
+     
+2. and how the tool would be built: 
+    * Syncup process from Filecoin Network to DB.
```
Comment (eshon) [inline rfp-proposals/rfp-proposal-reputation-system.md]: Hi @sahilnanda1995,

Question: What database do you intend to use?

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+Funding Required: $10,000 
+
+
+### Milestone 2: #buidl-ing the service 
+
+1. Setting up the sync progress from Filecoin Node to DB. 
+2. Writing API's + code for users to: 
+    * query miner metrics above.
+    * submit claims: 
+        * that get verified via data that is contextual to the claim.
```
Comment (eshon) [inline rfp-proposals/rfp-proposal-reputation-system.md]: Can you provide an example  of a claim that gets verified?

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+
+* Comparison with storage miners with the same score:
+    * Storage price
+    * Collateral
+    * Est. uptime
+    * Stored data
+
+* Substantiated feedback:
+    * identifying the user:
+        * a message-signing-based authentication mechanism with a user's public address as their identifier.
```
Comment (eshon) [inline rfp-proposals/rfp-proposal-reputation-system.md]: How will the user do this? One of you docs mentions Metamask but that doesn't exist for Filecoin.

File: rfp-proposals/rfp-proposal-reputation-system.md (hunk 1 above)
Comment (sahilnanda1995) [reply rfp-proposals/rfp-proposal-reputation-system.md]: Hi @eshon, 

At the moment, we intend to go forward with postgres. In my research, I came across the  ChainWatch tool which now seems to be configured with postgres as well.  Proposed path is thinking how much overlap exists so we can leverage existing work, and how much extra features we would need to add for the service to be complete in its feature set.

File: rfp-proposals/rfp-proposal-reputation-system.md (hunk 3 above)
Comment (sahilnanda1995) [reply rfp-proposals/rfp-proposal-reputation-system.md]: Yup, we also came to this conclusion and hence we were thinking of going with keybase until the time the community develops Filecoin Wallet. 

Thought process: 
If we do get funded, we see an overlap in terms of timelines that when get to the development of user auth module, we would likely be able to leverage the work being done by the team(s) that are parrellely developing wallets for Filecoin and work with them to integrate it in our service in time for mainnet launch.

File: rfp-proposals/rfp-proposal-reputation-system.md (hunk 2 above)
Comment (sahilnanda1995) [reply rfp-proposals/rfp-proposal-reputation-system.md]: Sure. One of the claims by clients could be that data was removed before the deal was supposed to expire. This is our speculation so far from my research that if a client loses data that they had stored with a particular miner, even though the protocol will slash the miner, the client may not know about the fact that the miner was slashed. This is how I imagine the service to work: 

1. Client submits claim.  
2. Claim get’s verified. Appropriate info is shown to the user. 
3. Once the claim is verified -> they can use the link to share it with the community. 

Additional context: As mentioned in milestone 1 of our proposal, our team will be deep diving into ConsensusFaults and StorageDealFaults to figure out possible claims clients can raise which can be verified from on chain activity.

**Comment by sahilnanda1995** [comment]
Hi @eshon, following on from our previous conversation. Happy to provide more answers if you have any more questions for me :)

**Comment by eshon** [comment]
> Hi @eshon, following on from our previous conversation. Happy to provide more answers if you have any more questions for me :)

Hi @sahilnanda1995 can you please email devgrants@filecoin.org and we can chat about options

**Comment by eshon** [comment]
Greetings,

Thank you for your application to the Filecoin DevGrants program. We appreciate your proposal!

Your team was not selected for this grant in Wave 2. We hope you apply for a dev grant in Wave 3 whose deadline is coming up March 31st and as discussed we'd be happy to chat with you about potential project ideas.

As the Filecoin implementations advance towards mainnet, we hope you continue to consider new proposals for projects. We would love to see more applications from you in the future.

Please stay tuned for more Filecoin dev grant RFPs.

**Pull request closed without merging**